artichoke-backend/src/extn/core/env/env.rb

```diff
@@ -1,3 +1,121 @@
 # frozen_string_literal: true
 
-ENV = EnvClass.new
+class EnvClass
+  def assoc(name)
+    value = self[name]
+
+    return nil if value.nil?
+
+    [name, self[name]]
```

Comment:
@cdeler this comment might have gotten lost in the last review. Can we replace the call to `self[name]` here with the value we already pulled out of the env?

I'm worried about a race as we call back into Rust as Artichoke gains support for multi-threading.

Reply:
Done